sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/retryprocessingerror/RetryProcessingErrorMLS.java

```diff
@@ -71,21 +85,26 @@ private void processRecordsByPartition(
           }
 
         } catch (RuntimeException e) {
-          LOGGER.error(
-              "Error while handling record {} in message handler {}",
-              consumerRecord.key(),
-              handler.getClass(),
-              e);
-          boolean shouldContinue = errorHandler.handleError(consumerRecord, e, consumer);
-          if (!shouldContinue) {
-            throw new StopListenerException(e);
+          long retryCount = getOffsetCounter(partition, consumerRecord.offset()).inc();
+          if (retryCount > maxRetryCount) {
+            LOGGER.error(
+                "Error while handling record {} in message handler {}, no more retries",
+                consumerRecord.key(),
+                handler.getClass(),
+                e);
+
+            callErrorHandler(finalErrorHandler, consumerRecord, e, consumer);
+            lastCommitOffset = markConsumerRecordProcessed(consumerRecord);
           } else {
             LOGGER.warn(
-                "Error while handling record {} in message handler {}, will be retried",
+                "Error while handling record {} in message handler {}, will be retried (%s / %s)..."
+                    .formatted(retryCount, maxRetryCount),
                 consumerRecord.key(),
                 handler.getClass(),
                 e);
 
+            callErrorHandler(intermediateErrorHandler, consumerRecord, e, consumer);
```

Comment:
If I understand this correctly, the error handler can not decide to stop consuming any more. Also the final [error handler](https://github.com/SDA-SE/sda-dropwizard-commons/pull/3922/files#diff-8fb386b215a69d009a70fdc9ae129fe6e8a350b2dc49dd46caeaaf286df31073R96) can only do something to mitigate the error but can't decide that something went very wrong and it's better to stop.

Reply:
I just moved the existing code into a method. Functionality stays the same:

```
private void callErrorHandler(...) {
    if (errorHandler != null) {
      boolean shouldContinue = errorHandler.handleError(consumerRecord, e, consumer);
      if (!shouldContinue) {
        throw new StopListenerException(e);
      }
    }
  }
```

Reply:
Ah no. It's moved to the callErrorHandler function.